meson.build

```diff
@@ -0,0 +1,277 @@
+    nasm_r = run_command(nasm, '-v', check: true)
+    out = nasm_r.stdout().strip().split()
+    if out[1].to_lower() == 'version'
+        if out[2].version_compare('< 2.10')
+            warning('nasm is too old (found @0@); ASM functions are disabled.'.format(out[2]))
+            enable_asm = false
+        endif
+    else
+        warning('Unexpected nasm version string: @0@; disabling.'.format(nasm_r.stdout()))
+        enable_asm = false
+    endif
+endif
+
+if get_option('asm').enabled() and (not enable_asm)
+    error('ASM functions were enabled but are unavailable.')
+elif (not enable_asm) and use_nasm and (not get_option('asm').disabled())
+    warning('Install nasm-2.10 or later for a significantly faster libass build.')
+endif
+
+# System check
+if enable_asm
+    if host_system == 'windows'
+        asm_args += ['-f', 'win' + bittype]
+        if bittype == '32'
+            asm_args += '-DPREFIX'
+        endif
+    elif host_system == 'darwin'
+        if cpu_family == 'arm64'
+            asm_args += '-DPREFIX'
+        else
+            asm_args += ['-f', 'macho' + bittype, '-DPREFIX', '-DSTACK_ALIGNMENT=16']
+        endif
+    elif host_system == 'linux' or host_system == 'sunos' or host_system == 'haiku' or host_system == 'gnu'
+        asm_args += ['-f', 'elf' + bittype, '-DSTACK_ALIGNMENT=16']
+    elif host_system == 'dragonfly' or host_system.endswith('bsd')
+        asm_args += ['-f', 'elf' + bittype]
+    else
+        error('Please contact libass upstream to figure out if ASM support for your platform can be added. In the meantime, disabling.')
+        enable_asm = false
+    endif
+endif
+
+if enable_asm
+    conf.set('CONFIG_ASM', 1)
+    if cpu_family == 'x86'
+        conf.set('ARCH_X86', 1)
+    elif cpu_family == 'arm64'
+        conf.set('ARCH_AARCH64', 1)
+    endif
+
+    if use_nasm
+        asm_args += '-Dprivate_prefix=ass'
+        asm_args += ['-I', meson.current_source_dir() / 'libass' + '/', '-DPIC=1', '-o', '@OUTPUT@', '@INPUT@']
+
+        nasm_gen = generator(nasm, output: '@BASENAME@.o', arguments: asm_args)
+    else
+        add_project_arguments(asm_args, language: 'c')
+    endif
+endif
+
+config_h_in = configure_file(output: 'config.h.in', configuration: conf)
+
+config_h = vcs_tag(command: ['git', 'describe', '--tags', '--long', '--always', '--broken', '--abbrev=40'],
+                   fallback: 'failed to determine (>= @0@)'.format(meson.project_version()),
+                   input: config_h_in,
+                   output: 'config.h')
+
+incs = include_directories('.', 'libass')
+
+subdir('libass')
+if get_option('test')
+    subdir('test')
+endif
+if get_option('profile')
+    subdir('profile')
+endif
+
+# libass.pc
+pkg = import('pkgconfig')
+pkg.generate(libass,
+             name: 'libass',
+             description: 'libass is an SSA/ASS subtitles rendering library')
```

Comment:
Could also add `meson.override_dependency('libass', libass_dep)`.

Reply:
@torque i have no idea what the `meson.override_dependency('libass', libass_dep)` suggestions does, can you look into it?

Reply:
It allows users who are including libass as a subproject in *their* project to use libass_dep automatically just by adding libass as a submodule. You can still externally map the libass_dep variable as the provider for `dependency('libass')` using the .wrap file syntax, but override_dependency is a bit more streamlined.